src/hotspot/share/opto/memnode.cpp

```diff
@@ -528,8 +528,13 @@ bool MemNode::detect_ptr_independence(Node* p1, AllocateNode* a1,
 }
 
 
-// Find an arraycopy that must have set (can_see_stored_value=true) or
-// could have set (can_see_stored_value=false) the value for this load
+// Find an arraycopy ac that produces the memory state represented by parameter mem.
+// Return ac if
+// (a) can_see_stored_value=true  and ac must have set the value for this load or if
+// (b) can_see_stored_value=false and ac could have set the value for this load or if
+// (c) can_see_stored_value=false and ac cannot have set the value for this load.
+// In (c) also change the parameter mem to the memory input of ac.
```

Comment:
I suggest to modify last line to something like to make it clear why we do that:
`In (c) case change the parameter mem to the memory input of ac to skip it when searching stored value.`

Reply:
Sure, I've done that now.